### ISSUE TITLE
messageReactionAdd: uncached user partial does not have a user ID set

### ISSUE THREAD

**Issue by Treeston** [discordjs/discord.js#5033]
**Please describe the problem you are having in as much detail as possible:**

My bot takes some action if a user reacts to its messages.
It has no privileged intents, and I only want to fetch users when they become relevant to me.

In order to allow this, the [guide](https://discordjs.guide/popular-topics/reactions.html) instructs me to add the `USER` partial to instantiation.

When I do so, `messageReactionAdd` fires. However, the `user` partial that is passed is invalid, as its `id` is undefined.
This, unsurprisingly, causes `user.fetch()` to result in a 404 error.

By monitoring `raw`, I have verified that the data sent by Discord indeed includes the user ID.

**Include a reproducible code sample here, if possible:**

To trigger the bug, use an uncached user to react to the (cached) test message.

```js
const Discord = new require('discord.js');
const config = require('./config.json');

const bot = new Discord.Client({ partials: ['CHANNEL', 'MESSAGE', 'REACTION', 'USER'] });
bot.on('messageReactionAdd', (reaction, user) => {
    console.log('messageReactionAdd user:', user);
});
bot.on('raw', ({t,d}) =>
{
    if (t === 'MESSAGE_REACTION_ADD')
        console.log('raw reaction user ID:',d.user_id);
});

(async () =>
{
    await bot.login(config.token);
    const channel = await bot.channels.fetch(config.pingChannel);
    await channel.send('test message');
})();
```

**Further details:**

- discord.js version: `discord.js@12.4.1`
- Node.js version: `v14.15.1`
- Operating system: `Windows 10`
- Priority this issue should have – please be realistic and elaborate if possible: `high`

**Relevant client options:**

- partials: `CHANNEL`, `MESSAGE`, `REACTION`, `USER`
- gateway intents: none
- other: none

- [ ] I have also tested the issue on latest master, commit hash:

**Comment by almostSouji** [comment]
fixed as of 12.5.0 which was released a few hours ago. (which included https://github.com/discordjs/discord.js/pull/4969)

**Comment by Treeston** [comment]
Apologies, didn't realize there was a new version on `npm` since last night.
